### ISSUE TITLE
C API: some functions are unimplemented but not documented

### ISSUE THREAD

**Issue by TerrorJack** [bytecodealliance/wasmtime#2895]
The following functions in C API are actually not implemented in rust, but not marked as unimplemented in the documentation. This may cause linker errors when downstream bindings try to consume the headers and emit foreign function declarations.

```
wasm_frame_copy
wasm_func_as_extern_const
wasm_global_as_extern_const
wasm_table_as_extern_const
wasm_memory_as_extern_const
```

**Comment by alexcrichton** [comment]
Thanks for the report! I'll look to update this documentation in https://github.com/bytecodealliance/wasmtime/pull/2897 as I touch a whole bunch of other docs too.

**Comment by alexcrichton** [comment]
I think this was fixed in #2897
